code/modules/food&drinks/kitchen machinery/servingplate.dm

```diff
@@ -0,0 +1,166 @@
+/obj/machinery/servingplate
+	name = "Serving Plate"
+	desc = "Keeps food at a desired temperature"
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "cooler"
+	density = 1
+	anchored = 1
+	use_power = 1
+	layer = 2.8
+	throwpass = 1
+	idle_power_usage = 15
+	active_power_usage = 150
+	var/freshModifier = 10 //the number to * or / when converting values
+	var/targetTemp = 0
+	var/cold = TRUE
+
+/obj/machinery/servingplate/process()
+	for(var/O in src.loc)
+		if(istype(O,/obj/item/weapon/reagent_containers/food))
+			var/obj/item/weapon/reagent_containers/food/F = O
+			if(cold)
+				if(F.freshMod*freshModifier > -targetTemp)
+					F.freshMod = max(-1,F.freshMod - (targetTemp/freshModifier))
+			else
+				if(F.freshMod*freshModifier < targetTemp)
+					F.freshMod = min(1,F.freshMod + (targetTemp/freshModifier))
+			if(emagged && !F.tainted)
+				F.tainted = TRUE
+			use_power(active_power_usage * targetTemp)
+
+/obj/machinery/servingplate/proc/fluffTemp(var/what)
+	if(what < 0)
+		what = -what
+	switch(what) //to is not inclusive, needs to be x-1 to y-1
+		if(-110 to 3)
+			return cold ? "Cool" : "Lukewarm"

```

Comment:
@Miauw62 No. It instantly says "boiling/chilling" when you switch, but it takes a moment for the temperature to go down or up. Cold is a negative number, hot is a positive.

If that's too confusing I can change it.